src/main/java/bio/terra/service/dataset/flight/ingest/IngestValidateRefsStep.java

```diff
@@ -39,4 +40,32 @@ public StepResult undoStep(FlightContext context) {
     // The update will update row ids that are null, so it can be restarted on failure.
     return StepResult.getStepResultSuccess();
   }
+
+  public static class InvalidRefId {
+    public String refId;
+    public String columnName;
```

Comment:
fields can be `final`
```suggestion
    public final String refId;
    public final String columnName;
```